File: MetodosOrdenamiento/bucket/bucket.py
import random

def bucket_sort(arr):
    if len(arr) == 0:
        return arr
    
    # Encontrar valor máximo y mínimo
    min_val = min(arr)
    max_val = max(arr)
    
    # Crear buckets
    bucket_count = len(arr)
    buckets = [[] for _ in range(bucket_count)]
    
    # Distribuir elementos en los buckets
    for num in arr:
        # Calcular índice del bucket
        index = int((num - min_val) / (max_val - min_val + 1) * bucket_count)
        buckets[index].append(num)
    
    # Ordenar cada bucket e insertar en el array original
    arr.clear()
    for bucket in buckets:
        bucket.sort()  # Puedes usar cualquier algoritmo de ordenación
        arr.extend(bucket)
    
    return arr

# Generar arreglo de 10 elementos con números aleatorios del 1 al 100
arr = [random.randint(1, 100) for _ in range(10)]
print("Arreglo original:", arr)

# Ordenar con bucket sort
arr_ordenado = bucket_sort(arr.copy())
print("Arreglo ordenado:", arr_ordenado)

File: MetodosOrdenamiento/bucket/bucket.js
function bucketSort(arr) {
    if (arr.length === 0) return arr;
    
    // Encontrar valor máximo y mínimo
    let minVal = Math.min(...arr);
    let maxVal = Math.max(...arr);
    
    // Crear buckets
    let bucketCount = arr.length;
    let buckets = Array.from({length: bucketCount}, () => []);
    
    // Distribuir elementos en los buckets
    for (let num of arr) {
        let index = Math.floor((num - minVal) / (maxVal - minVal + 1) * bucketCount);
        buckets[index].push(num);
    }
    
    // Ordenar cada bucket e insertar en el array original
    let result = [];
    for (let bucket of buckets) {
        bucket.sort((a, b) => a - b);
        result.push(...bucket);
    }
    
    return result;
}

// Generar arreglo de 10 elementos con números aleatorios del 1 al 100
let arr = Array.from({length: 10}, () => Math.floor(Math.random() * 100) + 1);
console.log("Arreglo original:", arr);

// Ordenar con bucket sort
let arrOrdenado = bucketSort([...arr]);
console.log("Arreglo ordenado:", arrOrdenado);